### PR TITLE
Lower `aten::round` operator

### PR DESCRIPTION
**Pull request by jysohn23**
Implements Bankers Rounding, where it round to the nearest even integer
if equidistant from 2 integers, matching `aten::round`.

Used in DLRM (cc: @taylanbil).

### CODE REVIEW

File: test/cpp/test_aten_xla_tensor.cpp [hunk 1]
```diff
@@ -2397,6 +2397,21 @@ TEST_F(AtenXlaTensorTest, TestFloor) {
...
+       // xla::RoundToEven properly implements bankers rounding.
+       torch::tensor({-0.5, 0.5}, torch::TensorOptions(torch::kFloat))},
+      0);
+  torch::Tensor b = torch::round(a);
+  ForEachDevice([&](const torch::Device& device) {
+    torch::Tensor xla_a = CopyToDevice(a, device);
+    torch::Tensor xla_b = torch::round(xla_a);
+    AllClose(b, xla_b);
+  });
+}
```
Comment (JackCaoG) [inline test/cpp/test_aten_xla_tensor.cpp:2413]: Could you add 
```
  ExpectCounterNotChanged("aten::.*", cpp_test::GetIgnoredCounters());                                                                                                                                                                                       
  ExpectCounterChanged("xla::round", cpp_test::GetIgnoredCounters()); 
```
after `ForEachDevice` to verified that `xla::round` is called?

File: torch_xla/csrc/aten_xla_type.h [hunk 2]
```diff
@@ -827,6 +827,10 @@ class AtenXlaType {
   static at::Tensor& resize_(at::Tensor& self, at::IntArrayRef size,
                              c10::optional<at::MemoryFormat> memory_format);
 
+  static at::Tensor round(const at::Tensor& self);
```
Comment (JackCaoG) [inline torch_xla/csrc/aten_xla_type.h:830]: Could you also add `round_out ` to https://github.com/pytorch/xla/blob/master/scripts/gen.py#L100? This way the out function will be redirect to use the xla implementation here. Example in https://github.com/pytorch/xla/pull/1592/files

File: test/cpp/test_aten_xla_tensor.cpp (hunk 1 above)
Comment (jysohn23) [reply test/cpp/test_aten_xla_tensor.cpp:2416]: Didn't see that assertion on other tests but did for others. Anyways added for this.

File: torch_xla/csrc/aten_xla_type.h (hunk 2 above)
Comment (jysohn23) [reply torch_xla/csrc/aten_xla_type.h:830]: Done, thanks.

**Pull request merged**